lib/chaotic_order/manager.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module ChaoticOrder
+  class Manager
+    attr_reader :arel, :adapter_name
+
+    def initialize(arel, adapter_name)
+      @arel = arel
+      @adapter_name = adapter_name
+    end
+
+    def set_random_order
+      return unless set_random_order?
+
+      arel.order(random_for_adapter)
+    end
+
+    private
+
+    def set_random_order?
+      !order_set? && !distinct_set?
+    end
+
+    def order_set?
+      arel.orders.present?
+    end
+
+    def distinct_set?
+      arel.to_sql =~ /SELECT DISTINCT/i
```

Comment:
Yeah, I thought about this solution, just wasn't sure if:

 - All db providers share that syntax
 - There is already a method within `ActiveRecord::Relation` that represents the existence of `distinct` clause
 
As a side comment, I'm not sure that is the right regex:
<img width="1039" alt="Screen Shot 2021-12-30 at 18 46 13" src="https://user-images.githubusercontent.com/50113670/147790116-a9522e50-0d11-4187-9a3a-296b1f919ca7.png">

Having said that, I haven't found any cleaner way of doing this 🤔

Reply:
yeah, I'm not so worried about it, worst case scenario we are not applying the random order for that query and that's it. That said, a way to improve this would be changing the regex to `/^SELECT DISTINCT/i` I'll do that